scripts/deploy.py

```diff
@@ -0,0 +1,497 @@
+    with open(CHANGELOG_PATH, 'r') as f:
+        contents = f.read()
+
+    contents = "v%s\n%s\n\nPLEASE EDIT THE FOLLOWING LOG!\n\n%s\n" % (
+        FormatVersion(ctx['new-version']),
+        '~' * (1 + len(FormatVersion(ctx['new-version']))),
+        ctx['git-log'],
+    ) + contents
+
+    with open(CHANGELOG_PATH, 'w') as f:
+        contents = f.write(contents)
+
+    return True
+
+
+def SaveGitLog(ctx):
+    """Store git log for appending to changelog.txt"""
+    proc = subprocess.Popen(
+        ['git', 'log',
+         'v%s..HEAD' % FormatVersion(ctx['old-version'])],
+        stdout=subprocess.PIPE)
+    ctx['git-log'] = proc.communicate()[0].decode('utf-8')
+    if proc.returncode != 0:
+        click.secho('Git returned non-zero exit code!', fg='red')
+        return False
+
+    return True
+
+
+def GetNextVersion(ctx):
+    """Select new major/minor version."""
+    v = ctx['old-version']
+    variants = [
+        {
+            'MAJOR': v['MAJOR'],
+            'MINOR': v['MINOR'] + 1,
+            'PATCH': 0,
+            'POSTFIX': 'rc1',
+        },
+        {
+            'MAJOR': v['MAJOR'] + 1,
+            'MINOR': 0,
+            'PATCH': 0,
+            'POSTFIX': 'rc1',
+        },
+    ]
+    ctx['new-version'] = SelectionPrompt(
+        "Current version is v%s. What will be the new one?" % FormatVersion(v),
+        [(FormatVersion(x), x) for x in variants])
+    ctx['dev-version'] = {
+        'MAJOR': ctx['new-version']['MAJOR'],
+        'MINOR': ctx['new-version']['MINOR'] + 1,
+        'PATCH': 0,
+        'POSTFIX': 'dev',
+    }
+    return True
+
+
+def GetNextPatchVersion(ctx):
+    """Select new patch version."""
+    v = ctx['old-version']
+    m = re.search(r'rc(\d+)', v['POSTFIX'])
+    if m:
+        variants = [
+            {
+                'MAJOR': v['MAJOR'],
+                'MINOR': v['MINOR'],
+                'PATCH': v['PATCH'],
+                'POSTFIX': 'rc%d' % (int(m.group(1)) + 1),
+            },
+            {
+                'MAJOR': v['MAJOR'],
+                'MINOR': v['MINOR'],
+                'PATCH': v['PATCH'],
+                'POSTFIX': '',
+            },
+        ]
+    else:
+        variants = [
+            {
+                'MAJOR': v['MAJOR'],
+                'MINOR': v['MINOR'],
+                'PATCH': v['PATCH'] + 1,
+                'POSTFIX': '',
+            },
+            {
+                'MAJOR': v['MAJOR'],
+                'MINOR': v['MINOR'],
+                'PATCH': v['PATCH'] + 1,
+                'POSTFIX': 'rc1',
+            },
+        ]
+    ctx['new-version'] = SelectionPrompt(
+        "Current version is v%s. What will be the new one?" % FormatVersion(v),
+        [(FormatVersion(x), x) for x in variants])
+    return True
+
+
+def ReadCurrentVerionFromFile(ctx):
+    """Read current version from version.inc file"""
+    version = {}
+    macro_re = re.compile(r'#define LC0_VERSION_(\S+) (.*)')
```

Comment:
*could* be placed outside this function, at global level as _hidden_variable. not a big deal either way. honestly, since it's only used a handful of times and not interspersed with other regexes, you could remove it entirely with no performance hit (re module automatically caches recently compiled regexes)